core/capabilities/remote/executable/parallel_executor_test.go

```diff
@@ -82,8 +84,15 @@ func Test_ExecutingMultipleTasksInParallel(t *testing.T) {
 
 func Test_StopsExecutingMultipleParallelTasksWhenClosed(t *testing.T) {
 	tp := newParallelExecutor(10)
-
 	var counter int32
+	t.Cleanup(func() {
+		assert.Eventually(t, func() bool {
+			return atomic.LoadInt32(&counter) == 0
+		}, 10*time.Second, 10*time.Millisecond)
```

Comment:
Why do we need an eventually? The service has been stopped at this point. Are there other background routines still running?